packages/metal-quartz-button/src/MetalQuartzButton.soy

```diff
@@ -0,0 +1,40 @@
+{namespace MetalQuartzButton}
+
+/**
+ * This renders the component's whole content.
+ */
+{template .render}
+	{@param? block: bool}
+	{@param? disabled: bool}
+	{@param? href: string}
+	{@param? icon: string}
+	{@param? label: html}
+	{@param? name: string}
+	{@param? size: string}
+	{@param? spritemap: string}
+	{@param? style: string}
+	{@param? type: string}
+	{@param? value: string}
+	{if $href}
+		<a href="{$href}" {call MetalQuartzButtonAttributes.attributes data="all" /}>
+			{call .iconMarkup data="all" /}
+			{$label or ''}
+		</a>
+	{else}
+		<button {call MetalQuartzButtonAttributes.attributes data="all" /}>
+			{call .iconMarkup data="all" /}
+			{$label or ''}
+		</button>
+	{/if}
+{/template}
+
+{template .iconMarkup}
```

Comment:
I think for our template naming convention, we should avoid using the work "markup", since all templates are realistically going to be returning some sort of markup. I would call this just `{template .icon}`.